control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -173,8 +173,48 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager, create
 	return nil
 }
 
-func (r *HostedControlPlaneReconciler) setup(createOrUpdate upsert.CreateOrUpdateFN) {
-	r.createOrUpdate = createOrUpdateWithOwnerRefFactory(createOrUpdate)
+func isScrapeConfig(obj client.Object) bool {
+	switch obj.(type) {
+	case *prometheusoperatorv1.ServiceMonitor:
+		return true
+	case *prometheusoperatorv1.PodMonitor:
+		return true
+	}
+
+	return false
+}
+
+func isClusterVersionAvailable(hcp *hyperv1.HostedControlPlane) bool {
+	for _, condition := range hcp.Status.Conditions {
```

Comment:
nit: pls use meta.FindStatusCondition()  as in:
https://github.com/openshift/hypershift/blob/a9eea639ad1cee72e10858c93c9e00d0a43f661c/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L508

Reply:
Thanks & agreed, I will use this function to factorise the code even more.

Speaking of the `ConditionType`s we use in `hostedcontrolplane_controller.go`; they are defined in [hostedcluster_conditions.go](https://github.com/openshift/hypershift/blob/a9eea639ad1cee72e10858c93c9e00d0a43f661c/api/v1beta1/hostedcluster_conditions.go#L26)?
Hence I hesitated using them in the first place because, well, the file in which they are defined seems to define `hostedcluster` conditions only.
Maybe we should rename `hostedcluster_conditions.go` to `conditions.go` as [it is the case](https://github.com/openshift/hypershift/blob/a9eea639ad1cee72e10858c93c9e00d0a43f661c/api/v1alpha1/conditions.go) in v1alpha1.

Reply:
Another thing which is quite disturbing is why does `FindStatusCondition` consume `string` and not `ConditionType` as parameter. The second argument is almost always converted from a `ConditionType` to a `string`.

Reply:
> Maybe we should rename hostedcluster_conditions.go to conditions.go as [it is the case](https://github.com/openshift/hypershift/blob/a9eea639ad1cee72e10858c93c9e00d0a43f661c/api/v1alpha1/conditions.go) in v1alpha1.

Agreed that would make sense.

> Another thing which is quite disturbing is why does FindStatusCondition consume string and not ConditionType as parameter. 

The function is from upstream kubernetes and does not know about our API's condition type.